Microsoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.

C:\Users\ccadenc>cd desktop/pythonstuff

C:\Users\ccadenc\Desktop\pythonstuff>cd PythonDjango

C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango>cd MyEnvironments

C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments>call djangoEnv/scripts/activate

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments>cd ..

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango>cd DojoNinjas

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py makemigrations
Migrations for 'dojo_ninjas':
  apps\dojo_ninjas\migrations\0001_initial.py
    - Create model Dojo
    - Create model Ninja

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dojo_ninjas, sessions
Running migrations:
  Applying dojo_ninjas.0001_initial... OK

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.create(name="Forest Dojo", city="Mesa", state=
... "AZ"
... )
<Dojo: Forest Dojo>
>>> dojo.objects.create(name="Rabbit Dojo", city="Tempe", state= "AZ")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'dojo' is not defined
>>> Dojo.objects.create(name="Rabit Dojo", city="Tempe", state="AZ")
<Dojo: Rabit Dojo>
>>> Dojo.objects.create(name="Aquarium Dojo", city="Scottsdale", state="AZ")
<Dojo: Aquarium Dojo>
>>> Dojo.objects.create(name="Tummy Dojo", city="Phoenix", state="AZ")
<Dojo: Tummy Dojo>
>>> Dojo.objects.get(id=4.dlete())
  File "<console>", line 1
    Dojo.objects.get(id=4.dlete())
                              ^
SyntaxError: invalid syntax
>>> Dojo.objects.get(id=4.delete())
  File "<console>", line 1
    Dojo.objects.get(id=4.delete())
                               ^
SyntaxError: invalid syntax
>>> b=Dojo.objects.get(id=2
...
^C
(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.all()
<QuerySet [<Dojo: Forest Dojo>, <Dojo: Rabit Dojo>, <Dojo: Aquarium Dojo>, <Dojo: Tummy Dojo>]>
>>> Dojo.objects.get(id=1)
<Dojo: Forest Dojo>
>>> Dojo.objects.get(id=1).delete()
(1, {u'dojo_ninjas.Ninja': 0, u'dojo_ninjas.Dojo': 1})
>>> dojo.objects.get(id=2)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'dojo' is not defined
>>> Dojo.objects.get(id=2).delete()
(1, {u'dojo_ninjas.Ninja': 0, u'dojo_ninjas.Dojo': 1})
>>> Dojo.objects.all()
<QuerySet [<Dojo: Aquarium Dojo>, <Dojo: Tummy Dojo>]>
>>> Dojo.objects.all()
<QuerySet [<Dojo: Aquarium Dojo>, <Dojo: Tummy Dojo>]>
>>> Ninja.objects.create(dojo=Dojo.objects.get(id=1), )
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 380, in get
    self.model._meta.object_name
DoesNotExist: Dojo matching query does not exist.
>>> Dojo.objects.get(id=1)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 380, in get
    self.model._meta.object_name
DoesNotExist: Dojo matching query does not exist.
>>> Dojo.objects.get(id=2)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 380, in get
    self.model._meta.object_name
DoesNotExist: Dojo matching query does not exist.
>>> Dojo.objects.get(id=0)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 380, in get
    self.model._meta.object_name
DoesNotExist: Dojo matching query does not exist.
>>> Dojo.objects.all()
<QuerySet [<Dojo: Aquarium Dojo>, <Dojo: Tummy Dojo>]>
>>> Dojo.objects.first
<bound method Manager.first of <django.db.models.manager.Manager object at 0x00000000045EBD68>>
>>> Dojo.objects.first()
<Dojo: Aquarium Dojo>
>>> Dojo.objects.first().id
3
>>> Dojo.objects.last().id
4
>>> Ninja.objects.create(dojo=Dojo.objects.get(id=3), first_name="Goldfish", last_name="Train")
<Ninja: Goldfish Train>
>>> Ninja.objects.create(dojo=Dojo.objects.get(id=4), first_name="Bunneh", last_name="Cha")
<Ninja: Bunneh Cha>
>>> Ninja.object.create(dojo=Dojo.objects.get(id=4), first_name="Chicken", last_name="Poopsicles")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: type object 'Ninja' has no attribute 'object'
>>> Ninja.objects.create(dojo=Dojo.objects.get(id=4), first_name="Chicken", last_name="Poopsicles")
<Ninja: Chicken Poopsicles>
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.all()
<QuerySet [<Dojo: Aquarium Dojo>, <Dojo: Tummy Dojo>]>
>>>
^C
(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> Dojo.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Dojo' is not defined
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 229, in __repr__
    return '<%s %r>' % (self.__class__.__name__, data)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\base.py", line 590, in __repr__
    u = six.text_type(self)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas\apps\dojo_ninjas\models.py", line 17, in __str__
    return "id = " + id + "name = " + self.name
TypeError: coercing to Unicode: need string or buffer, builtin_function_or_method found
>>>
^C
(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 229, in __repr__
    return '<%s %r>' % (self.__class__.__name__, data)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\base.py", line 590, in __repr__
    u = six.text_type(self)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas\apps\dojo_ninjas\models.py", line 17, in __str__
    return "id = " + self.id + "name = " + self.name
TypeError: coercing to Unicode: need string or buffer, int found
>>>
^C
(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from apps.dojo_ninjas.models import *
>>> Dojo.objects.all()
<QuerySet [<Dojo: name = Aquarium Dojo>, <Dojo: name = Tummy Dojo>]>
>>> Dojo.ojects.get(id=3)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: type object 'Dojo' has no attribute 'ojects'
>>> dojo.objects.get(id=3)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'dojo' is not defined
>>> Dojo.objects.get(id=3)
<Dojo: name = Aquarium Dojo>
>>> Ninja.objects.get(dojo="3")
<Ninja: name = Goldfish Train>
>>> Ninja.objects.get(dojo=4)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 384, in get
    (self.model._meta.object_name, num)
MultipleObjectsReturned: get() returned more than one Ninja -- it returned 2!
>>> Ninja.objects.get(dojo="4")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 384, in get
    (self.model._meta.object_name, num)
MultipleObjectsReturned: get() returned more than one Ninja -- it returned 2!
>>> Ninja.objects.get(dojo="4")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\MyEnvironments\djangoEnv\lib\site-packages\django\db\models\query.py", line 384, in get
    (self.model._meta.object_name, num)
MultipleObjectsReturned: get() returned more than one Ninja -- it returned 2!
>>>
Dojo.objects.first().ninjas.all()
Dojo.objects.last().ninjas.all()

(djangoEnv) C:\Users\ccadenc\Desktop\pythonstuff\PythonDjango\DojoNinjas>python manage.py makemigrations
You are trying to add a non-nullable field 'desc' to dojo without a default; we can't do that (the database needs something to populate existing rows).
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit, and let me add a default in models.py
Select an option: 1
Please enter the default value now, as valid Python
The datetime and django.utils.timezone modules are available, so you can do e.g. timezone.now
Type 'exit' to exit this prompt
>>> "empty"